File: src/Services/BudgetPlan/Andaha.Services.BudgetPlan/Requests/FixedCost/Dtos/V1/FixedCostHistoryDto.cs
using Andaha.Services.BudgetPlan.Infrastructure.EntityConfiguration;
using LinqKit;
using Microsoft.EntityFrameworkCore;
using System.Linq.Expressions;

namespace Andaha.Services.BudgetPlan.Requests.FixedCost.Dtos.V1;

public readonly record struct FixedCostHistoryDto(FixedCostDto FixedCost, DateTime ValidFrom, DateTime ValidTo)
{
    public static readonly Expression<Func<Core.FixedCost, FixedCostHistoryDto>> MapFromEntity =
        (fixedCost) => new FixedCostHistoryDto(
            FixedCostDto.MapFromEntity.Invoke(fixedCost),
            EF.Property<DateTime>(fixedCost, TemporalTable.ValidFrom),
            EF.Property<DateTime>(fixedCost, TemporalTable.ValidTo));
}

File: src/WebApps/MiniClient/Andaha.WebApps.MiniClient/src/app/api/budgetplan/dtos/FixedCostHistoryDto.ts
import { FixedCostDto } from "./FixedCostDto";

export interface FixedCostHistoryDto {
  fixedCost: FixedCostDto;
  validFrom: Date;
  validTo: Date;
}

export interface RawFixedCostHistoryDto {
  fixedCost: FixedCostDto;
  validFrom: string;
  validTo: string;
}
